### PR TITLE
Remove dependencies on OpenSSL

### PR DESCRIPTION
**Pull request by DemiMarie-parity**
We really really don’t want to depend on OpenSSL, as that requires us
make a new release for non-Linux platforms whenever an upstream OpenSSL
security vulnerability affects us.

We currently depend on OpenSSL only via the `native-tls` crate, which
we depend on via `hyper-tls` and `tokio-tls`.  This PR is a
work-in-progress because it only removes the first dependency.  Removing
the second one will require a change to the `websocket` crate.

### CODE REVIEW

File: core/offchain/src/api/http.rs [hunk 1]
```diff
@@ -554,9 +554,7 @@ enum WorkerToApi {
 /// Wraps around a `hyper::Client` with either TLS enabled or disabled.
 enum HyperClient {
```
Comment (tomaka) [inline core/offchain/src/api/http.rs]: We should remove the `HyperClient` enum entirely now.

**Comment by DemiMarie-parity** [comment]
@tomaka can you re-review?

File: core/offchain/src/api/http.rs [hunk 2]
```diff
@@ -47,12 +47,11 @@ pub fn http() -> (HttpApi, HttpWorker) {
...
 	};
 
+	let tls = hyper_rustls::HttpsConnector::new(1);
 	let engine = HttpWorker {
 		to_api,
 		from_api,
-		// TODO: don't unwrap; we should fall back to the HttpConnector if we fail to create the
-		// Https one; there doesn't seem to be any built-in way to do this
-		http_client: HyperClient::new(),
+		http_client: hyper::Client::builder().build(tls),
```
Comment (tomaka) [inline core/offchain/src/api/http.rs]:
```suggestion
		http_client: hyper::Client::builder().build(hyper_rustls::HttpsConnector::new(1)),
```

File: core/offchain/src/api/http.rs [hunk 3]
```diff
@@ -47,12 +47,11 @@ pub fn http() -> (HttpApi, HttpWorker) {
 		requests: FnvHashMap::default(),
 	};
 
+	let tls = hyper_rustls::HttpsConnector::new(1);
```
Comment (tomaka) [inline core/offchain/src/api/http.rs]:
```suggestion
```

**Comment by DemiMarie-parity** [comment]
@tomusdrw [`jsonrpsee` is not on `crates.io`](https://crates.io/crates/jsonrpsee)

**Pull request merged**